doc/developer/developing.rst

```diff
@@ -0,0 +1,285 @@
+.. _flycheck-developers-guide:
+
+=================
+Developer's Guide
+=================
+
+So you want to extend Flycheck, but have no idea where to start?  This guide
+will give you an overview of Flycheck internals, and take you through adding a
+syntax checker to Flycheck.
+
+An overview of Flycheck internals
+=================================
+
+The goal of Flycheck is to display errors from external checker programs
+directly in the buffer you are editing.  Instead of you manually invoking
+``make`` or the compiler for your favorite language, Flycheck takes care of it
+for you, collects the errors and displays them right there in the buffer.
+
+How Flycheck works is rather straightforward.  Whenever a syntax check is
+started (see :ref:`flycheck-syntax-checks`), the following happens:
+
+1. First, Flycheck runs the external program as an asynchronous process using
+   ``start-process``.  While this process runs, Flycheck simply accumulates its
+   output.
+2. When the process exits, Flycheck parses its output in order to collect the
+   errors.  The raw output is turned into a list of `flycheck-error` objects
+   containing, among others, the filename, line, column, message and severity of
+   the error.
+3. Flycheck then filters the collected errors to keep only the relevant ones.
+   For instance, errors directed at other files than the one you are editing are
+   discarded.
+4. Relevant errors are highlighted by Flycheck in the buffer, according to user
+   preference.  By default, each error adds a mark in the fringe at the line it
+   occurs, and underlines the symbol at the position of the error using
+   *overlays*.
+5. Finally, Flycheck rebuilds the error list buffer.
+
+Flycheck follows this process for all the :ref:`many different syntax checkers
+<flycheck-languages>` that are provided by default.
+
+.. note::
+
+   Specifically, the above describes the process of *command checkers*, i.e.,
+   checkers that run external programs.  All the checkers defined in
+   ``flycheck-checkers`` are command checkers, but command checkers are actually
+   instances of *generic checkers*.  See :flyc:`flycheck-ocaml` for an example
+   of how to use a generic checker.
+
+.. seealso::
```

Comment:
Oh neat!  I didn't know Github could preview RST.  Indeed, it shows a code block, but I've built the documentation locally and the block worked fine.